packages/providers/src/provider.ts

```diff
@@ -209,15 +211,34 @@ export default class Provider {
   operations: ReturnType<typeof getOperationsSdk>;
   cache?: MemoryCache;
 
+  consensusParamsCache?: ConsensusParameters;
+
   constructor(
     /** GraphQL endpoint of the Fuel node */
     public url: string,
     public options: ProviderOptions = {}
   ) {
+    // TODO: Disable or discourage the usage of this constructor in favour of connect()
     this.operations = this.createOperations(url, options);
     this.cache = options.cacheUtxo ? new MemoryCache(options.cacheUtxo) : undefined;
   }
 
+  static async connect(url: string, options: ProviderOptions = {}) {
+    const provider = new Provider(url, options);
+    const { consensusParameters } = await provider.getChain();
+    provider.consensusParamsCache = consensusParameters;
+    return provider;
+  }
+
+  async invalidateConsensusParamsCache() {
+    const { consensusParameters } = await this.getChain();
+    this.consensusParamsCache = consensusParameters;
+  }
```

Comment:
Since the consensus parameters will never change unless the URL that the `Provider` object is pointing to changes, I don't think we need to ever re-fetch them over an interval. Still, if a user wants to update them explicitly, I have exposed this method that lets them do so.

Reply:
Perhaps I'm overlooking something, but `consensusParameters` appears to me to be more of a class attribute rather than a cache. If that's the case, maybe we should consider removing the cache suffix.

I am under the impression that these `consensusParameters` should be fetched every time a Provider is instantiated with (`await Provider.connect('url')) and also whenever we update the Provider URL since a different provider would imply a different chain configuration. And only in these 2 cases.

Please let me know if I'm missing any details or if I'm oversimplifying things.

Reply:
I think it's cool to cache consensus parameters using the provider URL as the key. It can improve startup times if we are constructing new `Provider`s for the same URLs in different places of the application.

Reply:
By moving to this approach and making sure that a `provider` always has the `consensusParameter`, we get rid of the constants like `GAS_PRICE_FACTOR`. 

Since these values are going to be always present on the provider, we can easily access them on the `BaseInvocationScope`, `Account`, and `TransactionResponse` classes